tapeagents/observe.py

```diff
@@ -249,11 +249,16 @@ def dict_factory(cursor, row):
     cursor.close()
     calls: list[LLMCall] = []
     for row in rows:
+        output_dict = json.loads(row["output"])
+        # Handle logprobs separately before creating LLMOutput
+        if "logprobs" in output_dict:
```

Comment:
this is a part of LLMOutput validation, it does not belong here, it belongs in `LLMOutput` model definition. BTW, if `LLMOutput` is a Pydantic model, and `ChoiceLogprobs` is a Pydantic model, all the deserialized objects should have the correct types without this intervention.